paimon-core/src/main/java/org/apache/paimon/table/source/snapshot/TimeTravelUtil.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.table.source.snapshot;
+
+import org.apache.paimon.CoreOptions;
+import org.apache.paimon.Snapshot;
+import org.apache.paimon.options.Options;
+import org.apache.paimon.utils.SnapshotManager;
+import org.apache.paimon.utils.TagManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Contains time travel functions. */
+public class TimeTravelUtil {
+
+    public static List<String> scanKeys;
+
+    public static Snapshot resolveSnapshotFromOption(
+            Options options, SnapshotManager snapshotManager) {
+        List<String> scanHandleKey = new ArrayList<>();
+        for (String key : getScanKeys()) {
+            if (options.containsKey(key)) {
+                scanHandleKey.add(key);
+            }
+        }
+        
+        if (scanHandleKey.size() == 0) {
+            return null;
+        }
+        if (scanHandleKey.size() > 1) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "%s %s %s and %s can contains only one",
```

Comment:
Only one of the following parameters may be set  : [%s, %s, %s]

Reply:
Thanks @LinMingQiang had addressed